src/transformers/pipelines.py

```diff
@@ -1739,6 +1741,253 @@ def __call__(
+                )
+                conversation.set_history(cleaned_history[conversation_index])
+                output.append(conversation)
+            if len(output) == 1:
+                return output[0]
+            else:
+                return output
+
+    def _parse_and_tokenize(self, *args, **kwargs):
+        """
+        Parse arguments and tokenize, adding an EOS token at the end of the user input
+        """
+        # Parse arguments
+        inputs = self._args_parser(*args, **kwargs)
+        inputs = self.tokenizer.batch_encode_plus(inputs, add_special_tokens=False, padding=False).get("input_ids", [])
+        for input in inputs:
+            input.append(self.tokenizer.eos_token_id)
+        return inputs
+
+    def _clean_padding_history(self, generated_tensor) -> List[List[int]]:
+        """
+        Cleans the padding history. Padding may be generated in two places when multiple conversations are provided as
+        an input:
+            - at the end of the concatenated history and new user input, so that all input to the model have the same
+                length
+            - at the end of the generated response, as some responses will be longer than others
+        This method cleans up these padding token so that the history for each conversation is not impacted by the
+        batching process.
+        """
+        outputs = []
+        for sequence in generated_tensor:
+            sequence_tokens = []
+            is_previous_pad = False
+            for token in sequence:
+                if token == self.pad_token_id:
+                    if is_previous_pad:
+                        continue
+                    else:
+                        is_previous_pad = True
+                else:
+                    is_previous_pad = False
+                if self.framework == "pt":
+                    sequence_tokens.append(token.item())
+                else:
+                    sequence_tokens.append(int(token.numpy()))
+
+            outputs.append(sequence_tokens)
+        return outputs
+
+    def _concat_inputs_history(self, inputs: List[List[int]], histories: List[Optional[List[int]]], max_length: int):
+        """
+        Builds an input prepended by the history for this conversation, allowing multi-turn conversation with context
+        """
+        outputs = []
+        for input, history in zip(inputs, histories):
+            if history is not None:
+                input = history + input
+            if len(input) > max_length - self.min_length_for_response:
+                cutoff_eos_index = 0
+                while len(input) - cutoff_eos_index > max_length - self.min_length_for_response:
+                    cutoff_eos_index = input[cutoff_eos_index:].index(self.tokenizer.eos_token_id)
+                    if cutoff_eos_index == 0 or cutoff_eos_index == len(input) - 1:
+                        break
+                    else:
+                        input = input[cutoff_eos_index + 1 :]
```

Comment:
This should also break when the cutoff_eos_index is larger than the length of the remaining input. Otherwise it fails because `input[cutoff_eos_index:]` returns an empty list, which cannot be indexed with `self.tokenizer.eos_token_id`.

An easy fix is the following, which could probably be made cleaner.

```suggestion
                    if cutoff_eos_index >= len(input):
                        break
                        
                    cutoff_eos_index = input[cutoff_eos_index:].index(self.tokenizer.eos_token_id)
                    if cutoff_eos_index == 0 or cutoff_eos_index == len(input) - 1:
                        break
                    else:
                        input = input[cutoff_eos_index + 1 :]
```